src/utils/daos_hdlr.c

```diff
@@ -1953,6 +2326,16 @@ dm_connect(struct cmd_args_s *ap,
 				D_GOTO(err, rc);
 			}
 		}
+
+		/* do not copy a container that has unhealthy container status */
+		rc = dm_check_cont_status(ap, ca->src_coh, &status_healthy);
+		if (rc != 0) {
+			DH_PERROR_DER(ap, rc, "Failed to check container status");
+			D_GOTO(err, rc);
+		} else if (!status_healthy) {
+			DH_PERROR_DER(ap, rc, "Container status is unhealthy, stopping");
+			D_GOTO(err, rc);
+		}
```

Comment:
rc will be 0 here, so DH_PERROR_DER is wrong, perhaps you need to assign it.
```suggestion
                         rc = -DER_INVAL;
			DH_PERROR_DER(ap, rc, "Container status is unhealthy, stopping");
			D_GOTO(err, rc);
		}
```